### DISCUSSION TITLE
eoAPI_Maxar_demo: raster endpoint

### DISCUSSION THREAD

**Discussion by nsshah1288** [developmentseed/eoAPI#201] [General]
Hi, thank you so much for the demo located here: https://github.com/developmentseed/eoAPI/blob/main/demo/Maxar/eoAPI_Maxar_demo.ipynb 

it is very helpful! 

One small comment, when I was doing the Asset visualization, when I did `info = httpx.get(f"{raster_endpoint}/collections/{collection_id}/items/{item_id}/info").json()` , I was getting the following error: 
{
  "detail": "AWS_SECRET_ACCESS_KEY and AWS_NO_SIGN_REQUEST configuration options not defined, and /root/.aws/credentials not filled"
}

In the docker-compose.yml file [here](https://github.com/developmentseed/eoAPI/blob/main/docker-compose.yml), I removed the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables from titiler-pgstac and added `AWS_NO_SIGN_REQUEST=true`. Then, the raster endpoint worked to return the metadata for the assets. 

Thank you again for the demo!

**Comment by nsshah1288** [comment]
one more thing I forgot, the paging to get all the items was not working for me. I think it was not properly hitting the next link and therefore was requesting the same set of items over and over again. 

I changed the cell to look like the following, using a counter:
```
kahramanmaras_items = []

url = f"{stac_endpoint}/collections/{collection_id}/items"
x = 0
while True:
    if x == 0:
        items = httpx.get(url, params={"limit": 100}).json()
    else:
        items = httpx.get(url).json()
    
    kahramanmaras_items.extend(items["features"])
    next_link = list(filter(lambda link: link["rel"] == "next", items["links"]))
    if next_link:
        url = next_link[0]["href"]
        print(url)
    else:
        break
    x += 1    

print(f"Nb Items: {len(kahramanmaras_items)}")
```

and then it worked and returned 2115 items. I will make a small PR for this if I get some time

**Comment by vincentsarago** [comment]
@nsshah1288 🙏 thanks 

Would you be able to make a PR to update the notebook? 🙏

**Comment by jordanp-dluhc** [comment]
@vincentsarago just came across another thing which I'm hoping you might be able to help?

When trying to visualise the first asset:

```python
tilejson = httpx.get(
    f"{raster_endpoint}/collections/{collection_id}/items/{item_id}/tilejson.json",
    params=(
        ("assets", "visual"),  # THIS PARAMETER IS MANDATORY
        ("minzoom", 12),  # By default the tiler will use 0
        ("maxzoom", 19),  # By default the tiler will use 24
    ),
).json()
print(tilejson)

bounds = tilejson["bounds"]
m = ipyleaflet.leaflet.Map(
    center=((bounds[1] + bounds[3]) / 2, (bounds[0] + bounds[2]) / 2), zoom=12
)

geo_json = ipyleaflet.leaflet.GeoJSON(
    data=item, style={"opacity": 1, "dashArray": "9", "fillOpacity": 0.0, "weight": 4}
)
m.add_layer(geo_json)

tiles = ipyleaflet.leaflet.TileLayer(
    url=tilejson["tiles"][0],
    min_zoom=tilejson["minzoom"],
    max_zoom=tilejson["maxzoom"],
    bounds=[
        [bounds[1], bounds[0]],
        [bounds[3], bounds[2]],
    ],
)

m.add_layer(tiles)

m
```

I get the following error: `{'detail': 'Not Found'}`

Seems as if the `tilejson.json` file just doesn't exist in the raster endpoint.

Any ideas what might be causing that?

**Reply by hrodmn** [reply]
> I get the following error: `{'detail': 'Not Found'}`
> 
> Seems as if the `tilejson.json` file just doesn't exist in the raster endpoint.
> 
> Any ideas what might be causing that?

Hi @nsshah1288 - in `titiler==0.19` we introduced a [breaking change](https://github.com/developmentseed/titiler/blob/main/CHANGES.md#0190-2024-11-07) that might be causing this error. Try replacing the request URL with this:
```python
f"{raster_endpoint}/collections/{collection_id}/items/{item_id}/WebMercatorQuad/tilejson.json",
```

**Reply by jordanp-dluhc** [reply]
@hrodmn  thanks Henry! Now I get the following error:

`{"detail":"assets must be defined either via expression or assets options."}`

**Reply by jordanp-dluhc** [reply]
@hrodmn ignore me, I'm just being dumb - all seems to be working great now. 

@vincentsarago FYI - happy to submit a PR to incorporate @nsshah1288 's changes if that hasn't already been done? 

Also just to say thank you for all the work you all have put into the examples, documentation etc., makes it really easy to get up and running.
